ui/src/ducks/app/volumes.hooks.js

```diff
@@ -20,9 +20,34 @@ export function useRefreshVolume() {
     },
   );
 
+  const { data } = result;
+
   useEffect(() => {
     if (data) {
       dispatch(setVolumesAction(data));
     }
   }, [data, dispatch]);
+
+  return result;
+}
+
+export function useFetchCurrentVolumeObject(volumeName: string) {
```

Comment:
I'm not sure refactoring using "simple" Fetch hooks is the right approach. I would more see what is the expected data output, what is the input and then build a hook abstracting away fetching, data conversion and formatting. Kind of what I tried to initialize with the `useVolumesWithAlerts` hook.